src/Compilers/Core/Portable/Symbols/CompilerFeatureRequiredHelpers.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Reflection.Metadata;
+
+namespace Microsoft.CodeAnalysis;
+
+[Flags]
+internal enum CompilerFeatureRequiredFeatures
+{
+    None = 0,
+    RefStructs = 1 << 0,
+    RequiredMembers = 1 << 1,
+
+    Unknown = 1 << 31
+}
+
+internal static class CompilerFeatureRequiredHelpers
+{
+    internal static string? GetUnsupportedCompilerFeature(EntityHandle token, PEModule containingModule, CompilerFeatureRequiredFeatures allowedFeatures)
+    {
+        if (token.IsNil)
+        {
+            return null;
+        }
+
+        Debug.Assert((allowedFeatures & CompilerFeatureRequiredFeatures.Unknown) == 0);
+
+        var compilerRequiredFeatures = containingModule.GetCompilerFeatureRequiredFeatures(token);
```

Comment:
>GetCompilerFeatureRequiredFeatures [](http://example.com/codeflow?start=56&length=34)

It looks like this function is the only place where this helper is used. We can avoid building all the arrays then. I think it is worth doing. As I mentioned earlier, this function can be moved to PEModule.